spec/controllers/password_resets_controller_spec.rb

```diff
@@ -0,0 +1,65 @@
+require 'rails_helper'
+
+describe PasswordResetsController do
+  describe "GET show" do
+    it "renders show template if token is valid" do
+      alice = Fabricate(:user)
+      alice.update_column(:token, '123')
+      alice.save
+      get :show, id:'123'
+      expect(response).to render_template :show
+    end
+
+    it "sets @token" do
+      alice = Fabricate(:user)
+      alice.update_column(:token, '123')
+      alice.save
+      get :show, id:'123'
+      expect(assigns(:token)).to eq('123')
+    end
+
+    it "redirects to expired token page if token is invalid" do
+      get :show, id: 'wrongtoken123'
```

Comment:
The `:id` parameter feels kinda wrong, i would use custom route with something like `/password_reset/:token` where the `:token` would be fetched in params like `params[:token]` and would act totally like the id when building the named route in the email.